examples/example-all-fail/expected_results.json

```diff
@@ -0,0 +1,28 @@
+{
+  "version" : "v2",
+  "status" : "fail",
+  "message" : "Got 'not $A' instead of $A.",
+  "tests" : [
+    {
+      "name" : "Exercise method a",
+      "status" : "fail",
+      "message" : "Got 'not $A' instead of $A.",
+      "output" : "ExampleAllFailTest(TestAsserter)>>assert:description:resumable:\nExampleAllFailTest(TestAsserter)>>assert:description:\nExampleAllFailTest(TestAsserter)>>assert:equals:\nExampleAllFailTest>>testExerciseMethodA\nExampleAllFailTest(TestCase)>>performTest\n",
```

Comment:
The `output` key should only be used for user output. This looks like it wasn't user output. If it isn't and it is helpful to show to the user, it could be part of the `message` field.

See [the docs](https://exercism.org/docs/building/tooling/test-runners/interface):

> The per-test output key should be used to store and output anything that a user deliberately outputs for a test.

This comment also applies to the other files.

Reply:
I'm not really sure I understand purpose of output. So it is used for something, when test method prints out something to user? Any examples? I've included a stack-trace (as a debugging info) of runned test there, if method fails or raises error/exception.

Reply:
The purpose of output is to allow students to do print-like "debugging". It's sole purpose is for the student to output things that are then shown in the online editor. Note that most test runners don't support this, as capturing only what a student's code outputs is oftentimes tricky.

> I've included a stack-trace (as a debugging info) of runned test there, if method fails or raises error/exception.

This is great, but should just be part of the `message` field's contents.

Reply:
Ok, I will move stack-trace info (along with exception message text) into "message" field. I'd probably ignore user-defined output, since usually Smalltalk image doesn't use stdout to print output (it uses its own stream called Transcript). There could be a way that Transcript output would be re-directed and captured to "output" parameter, but it would be hard to distinguish things that are belonging just to executed test (There is no way to limit unrelated code to use Transcript as well).

Reply:
If it's hard, don't bother with it